### PR TITLE
Qshell cherry

### PR DESCRIPTION
**Pull request by toolslive**
inserting a compatibility layer, and make it work again with the qshell.
(CI likes it)

### CODE REVIEW

**Comment by domsj** [comment]
clusterName/clusterId isn't entirely correct yet. (just do a quick search for clusterName in files changed tab and you'll see where).

pull request can't be automatically merged. maybe merge latest 1.7 in here already to resolve conflicts? (or you could have better ideas :))

File: pylabs/extensions/arakoon_ext/client/ArakoonClient.py [hunk 1]
```diff
@@ -228,34 +219,39 @@ def getClientConfig (self, clusterId, configName = None):
...
-            return q.system.fs.joinPaths( clusterDir, "%s_client" % clusterId)
+            last = "%s_client" % clusterId
         else:
-            return q.system.fs.joinPaths( clusterDir, "%s_client_%s" % (clusterId, configName))
+            last = "%s_client_%s" % (clusterId, configName)
+
+        return '/'.join([clusterDir, last])
+
 
+        clientsCfg = X.getConfig( "arakoonclients")
```
Comment (domsj) [inline pylabs/extensions/arakoon_ext/client/ArakoonClient.py]: this won't do a lot (given the return a few lines above)

File: pylabs/extensions/arakoon_ext/server/ArakoonManagement.py [hunk 2]
```diff
@@ -784,16 +715,16 @@ def setTLSCertificate(self, node, cert_path, key_path):
...
+        if config.has_option(node, tls_key):
+            config.remove_option(node, tls_key)
 
-        config.addParam(node, tls_cert, cert_path)
-        config.addParam(node, tls_key, key_path)
+        config.set(node, tls_cert, cert_path)
+        config.set(node, tls_key, key_path)
 
-        config.write()
+        X.writeConfig(config, fn)
```
Comment (domsj) [inline pylabs/extensions/arakoon_ext/server/ArakoonManagement.py]: there seems to be a pattern here.
introduce a helper method so you can write `self._writeConfig(config)` instead?

File: pylabs/test/server/right/system_tests_anomaly.py [hunk 3]
```diff
@@ -22,62 +22,49 @@
...
-        try :
-            cmd = "umount %s" % mount_target
-            Common.run_cmd ( cmd )
-        except :
-            pass
-        fs.removeDirTree( mount_target )
+    if X.fileExists( mount_target ) :
+        Common.stopOne( Common.node_names[node_index] )
+        cmd = ["umount", mount_target]
+        X.subprocess.check_call ( cmd )
```
Comment (domsj) [inline pylabs/test/server/right/system_tests_anomaly.py:43]: Hmm, in the above there was some error handling. (Not of the best kind, but still it was there...)
Did you remove it on purpose?

Comment (toolslive) [inline pylabs/test/server/right/system_tests_anomaly.py:43]: yes. throwing away a 

```
try:
   ....
except:
   pass
```

is a good thing

File: pylabs/extensions/arakoon_ext/server/ArakoonManagement.py (hunk 2 above)
Comment (toolslive) [inline pylabs/extensions/arakoon_ext/server/ArakoonManagement.py]: yup. it wins 1 line per occurrence.

File: pylabs/extensions/arakoon_ext/client/ArakoonClient.py [hunk 4]
```diff
@@ -185,77 +176,75 @@ def _getClientConfig(clusterName, configName = None):
...
-                port = cfgFile.getValue(node, "client_port") 
-                node_dict.update({node: (ip_list,port)})
-            cluster_id = cfgFile.getValue("global", "cluster_id")
-            config = ArakoonClientConfig(cluster_id, node_dict)
+                ips = cfgFile.get(node, "ip")
+                ip_list = ips.split(',')
+                port = cfgFile.get(node, "client_port")
+                ip_port = (ip_list, port)
+                node_dict.update({node: ip_port})
+            config = ArakoonClientConfig(clusterName, node_dict)
```
Comment (domsj) [inline pylabs/extensions/arakoon_ext/client/ArakoonClient.py]: This used to be clusterId

**Pull request merged**